### PR TITLE
Python config classes for different pileup scenarios.

### PR DESCRIPTION
**Pull request by vvolkl**
Add an easier way to set pileup parameters globally without recompiling.

### CODE REVIEW

**Comment by jlingema** [comment]
test this please

File: Generation/options/pileup_hepmcreader.py [hunk 1]
```diff
@@ -6,18 +6,12 @@
...
 
-# assuming mm
-smeartool = FlatSmearVertex(
-    xVertexMin = -0.5,
-    xVertexMax = 0.5,
-    yVertexMin = -0.5,
-    yVertexMax = 0.5,
-    zVertexMin = -35,
-    zVertexMax = 35 )
+from FCCPileupScenarios import FCCPhase1Pileup
```
Comment (hegner) [inline Generation/options/pileup_hepmcreader.py]: no imports in the middle of code.

File: Generation/python/FCCPileupScenarios.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    """Beam Parameters Common to all FCC scenarios"""
+    def __init__(self, name):
+        self._name = name
+        self.VertexSmearingConfig = {
+          "xVertexMin": -0.5,
+          "xVertexMax": 0.5,
+          "yVertexMin": -0.5,
+          "yVertexMax": 0.5,
+          "zVertexMin": -35,
+          "zVertexMax": 35
```
Comment (hegner) [inline Generation/python/FCCPileupScenarios.py]: Could you add a reference to the paper where you took the parameters from?

File: Generation/options/pileup_hepmcreader.py [hunk 3]
```diff
@@ -6,18 +6,12 @@
...
+pileupscenario = FCCPhase1Pileup('PileupScenario')
+print pileupscenario.PileupToolConfig
```
Comment (hegner) [inline Generation/options/pileup_hepmcreader.py]: I'd not have printouts per default.

Comment (vvolkl) [inline Generation/options/pileup_hepmcreader.py]: Oh. That was for debugging and slipped through the cracks.

**Comment by vvolkl** [comment]
Tried to address the comments. Many thanks Benedikt! Also moved the filename property back to the config file, since that is not strictly related to pileup. Also at some point, the beamspot shape will have to be specified here.

File: Generation/python/FCCPileupScenarios.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
+
+from GaudiKernel import SystemOfUnits as units
+
+
+class FCCPileupScenarioBase(object):
+    """Beam Parameters Common to all FCC scenarios. Taken from
+    Benedikt, Michael, Daniel Schulte, and Frank Zimmermann:
```
Comment (hegner) [inline Generation/python/FCCPileupScenarios.py]: Michael Benedikt, Daniel Schulte, and Frank Zimmermann

File: Generation/python/FCCPileupScenarios.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+          "xVertexMin": -0.5 * units.mm,
+          "xVertexMax": 0.5 * units.mm,
+          "yVertexMin": -0.5 * units.mm,
+          "yVertexMax": 0.5 * units.mm,
+          "zVertexMin": -35 * units.mm,
+          "zVertexMax": 35 * units.mm
+        }
+        self.PileupToolConfig = {}
+
+class FCCPhase1Pileup(FCCPileupScenarioBase):
```
Comment (hegner) [inline Generation/python/FCCPileupScenarios.py]: Why two subclasses and not just two instances of a pile up scenario class?

Comment (vvolkl) [inline Generation/python/FCCPileupScenarios.py]: Makes a lot more sense -- I went one step further and used simple dictionaries instead (for now).

**Comment by hegner** [comment]
retest this please

**Pull request merged**